### ISSUE TITLE
`scx_lavd` seems to choke under either building this package with -j16, or building an LTO package that uses all 16 cores

### ISSUE THREAD

**Issue by kode54** [sched-ext/scx#223]
As is the Arch default now, everything builds with LTO. Building the Wayfire compositor causes heavy LTO build activity to fill all MAKEOPTS jobs (-j16, matching core count), and makes various things bog down, including Discord video calls, and if I'm using Wayfire, the compositor itself bogs down even under less stressful 16 core builds.

**Comment by multics69** [comment]
Thank you for reporting the problem. I am working on a better preemption scheme under scx_lavd. Hopefully, a new PR (#224 ) that was just sent partly addresses your problem. More optimization will come. Stay tuned.

**Comment by kode54** [comment]
Something wonky is going on with Github's commit history for that PR, I can't apply the five patches in order either with `patch -Np1` or with `git apply` or `git am`. I have to check out the branch directly, `git diff` it against `main`, then apply the patch to `main` in my build process. Will report on how it works out.

**Comment by multics69** [comment]
Thanks a lot!

**Comment by multics69** [comment]
If you don't mind, could you run `scx_lavd` with the `-s $(nproc)` option and share the log while your system is unusable for just 4-5 minutes? The log will contain the statistics of how `scx_lavd` works.

**Comment by multics69** [comment]
Thanks a lot!

**Comment by multics69** [comment]
@kode54 Could you please test if the problem still exists? PR #274 should fixed the problem.

**Comment by kode54** [comment]
I'll have to build a 6.9-rc7 kernel to test it, since 6.8.9 is unsupported.

**Comment by kode54** [comment]
It stops some of the stuttering, but it doesn't stop lag building up behind OBS and my camera input. The lag even shows in OBS itself, not just the client pulling from the virtual camera.

**Comment by multics69** [comment]
@kode54 Thanks for building the kernel and testing lavd. BTW, what is OBS? I will try to re-produce it on my end.

**Comment by multics69** [comment]
@kode54 If you don't mind, could you please share the LAVD logs?

**Comment by kode54** [comment]
Here's a log that stopped when I stopped the service shortly after the lag. OBS is OBS Studio, or Open Broadcast Software. https://obs-studio.org/

[scx_lavd.log.txt](https://github.com/sched-ext/scx/files/15281368/scx_lavd.log.txt)

**Comment by multics69** [comment]
@kode54   -- Hmm... I tried to reproduce the problem but it seems okay from my end.

The environment that I tested as follows:
   - 8-core AMD Ryzen 9 PRO 6950H with Radeon Graphics (16 cores with hyperthreading)

I ran the following workloads with `scx_lavd`:
  - full kernel build with defconfig and -j128 => around 4:30~4:40 (in an expected range)
  - glxgears: around 59 FPS => (in an expected range)
  - YouTube music streaming: no audio cracking and video jittering
  - OBS for video recording with an Intel integrated camera

Any hints so I can reproduce the problem? Is there any possibility that OBS is buggy?
